apm-agent-plugins/apm-error-logging-plugin/src/main/java/co/elastic/apm/agent/errorlogging/AbstractLoggerErrorCapturingInstrumentation.java

```diff
@@ -42,49 +42,39 @@
 
 public abstract class AbstractLoggerErrorCapturingInstrumentation extends TracerAwareInstrumentation {
 
-    @SuppressWarnings({"WeakerAccess"})
-    @VisibleForAdvice
-    public static final ThreadLocal<Boolean> nestedThreadLocal = new ThreadLocal<Boolean>() {
-        @Override
-        protected Boolean initialValue() {
-            return Boolean.FALSE;
-        }
-    };
+    public static final GlobalThreadLocal<Boolean> nestedThreadLocal = GlobalThreadLocal.get(AbstractLoggerErrorCapturingInstrumentation.class, "nested", Boolean.FALSE);
```

Comment:
Can be simplified with `co.elastic.apm.agent.sdk.state.CallDepth`. See `co.elastic.apm.agent.servlet.RequestStreamRecordingInstrumentation.GetInputStreamAdvice` as an example.

When using it, make sure to add `onThrowable = Throwable.class` to `OnMethodExit`. See also the Javadocs of `co.elastic.apm.agent.sdk.state.CallDepth#decrement` for more details.

Reply:
replaced with `CallDepth`